src/Common/EnvironmentVariableNames.cs

```diff
@@ -12,5 +17,50 @@ static class EnvironmentVariableNames
         public static readonly string WORKLOAD_UPDATE_NOTIFY_INTERVAL_HOURS = "DOTNET_CLI_WORKLOAD_UPDATE_NOTIFY_INTERVAL_HOURS";
         public static readonly string WORKLOAD_DISABLE_PACK_GROUPS = "DOTNET_CLI_WORKLOAD_DISABLE_PACK_GROUPS";
         public static readonly string TELEMETRY_OPTOUT = "DOTNET_CLI_TELEMETRY_OPTOUT";
+        public static readonly string DOTNET_ROOT = "DOTNET_ROOT";
+
+        private static readonly Version s_version6_0 = new(6, 0);
+
+        public static string? TryGetDotNetRootVariableName(string runtimeIdentifier, string defaultAppHostRuntimeIdentifier, string targetFrameworkVersion)
+            => TryGetDotNetRootVariableName(runtimeIdentifier, defaultAppHostRuntimeIdentifier, TryParseTargetFrameworkVersion(targetFrameworkVersion));
+
+        public static string? TryGetDotNetRootVariableName(string runtimeIdentifier, string defaultAppHostRuntimeIdentifier, Version? targetFrameworkVersion)
+        {
+            var targetArchitecture = TryParseArchitecture(runtimeIdentifier) ?? TryParseArchitecture(defaultAppHostRuntimeIdentifier);
+            if (targetArchitecture == null || targetArchitecture == RuntimeInformation.ProcessArchitecture)
+            {
+                var suffix = targetFrameworkVersion != null && targetFrameworkVersion >= s_version6_0 ?
+                    $"_{RuntimeInformation.ProcessArchitecture.ToString().ToUpperInvariant()}" :
+                    Environment.Is64BitProcess ? "" : "(x86)";
+
+                return DOTNET_ROOT + suffix;
+            }
+
+            return null;
+        }
+
+        private static Architecture? TryParseArchitecture(string runtimeIdentifier)
+        {
+            int separator = runtimeIdentifier.LastIndexOf("-", StringComparison.InvariantCulture);
+            if (separator < 0)
+            {
+                return null;
+            }
+
+            return runtimeIdentifier.Substring(separator + 1).ToLowerInvariant() switch
+            {
+                "arm" => Architecture.Arm,
+                "arm64" => Architecture.Arm64,
+                "x64" => Architecture.X64,
+                "x86" => Architecture.X86,
+                _ => null
+            };
```

Comment:
avoid maintaining list of supported architectures supported by using the enum provided by runtime:
```suggestion
            string architecture = runtimeIdentifier.Substring(separator + 1).ToLowerInvariant();
            return Enum.TryParse<Architecture>(architecture, true, out _) ? architecture : null;
```

Reply:
AFAIK, there is no guarantee that the names of the enum fields match the runtime identifier.

Reply:
dotnet host also parses the string using strongly typed enum: https://github.com/dotnet/runtime/blob/d5e3a5c2ca46691d65c81d520cb95f13f7a94652/src/installer/managed/Microsoft.NET.HostModel/Bundle/TargetInfo.cs#L26 so it is not possible to do anything useful with unknown architecture by the time control reaches this method.

Reply:
Thanks for the pointer. The code is parsing the runtime identifier like so: https://github.com/dotnet/runtime/blob/9d6396deb02161f5ee47af72ccac52c2e1bae458/src/installer/tests/Microsoft.NET.HostModel.Tests/Helpers/BundleHelper.cs#L136

Reply:
i doubt bundle tests run anywhere other than those few ci machines, but the source of truth is this public enum: https://github.com/dotnet/runtime/blob/7a45201181d37329b7ab0bee541ed7c42b5d94b6/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Architecture.cs#L9 and it is only growing